app/server.go

```diff
@@ -383,10 +385,19 @@ func NewServer(options ...Option) (*Server, error) {
 		}
 	}
 
-	if htmlTemplateWatcher, err2 := utils.NewHTMLTemplateWatcher("templates"); err2 != nil {
-		mlog.Error("Failed to parse server templates", mlog.Err(err2))
+	templatesDir, ok := fileutils.FindDir("templates")
+	if !ok {
+		mlog.Error("Failed find server templates", mlog.String("directory", "templates"))
 	} else {
-		s.htmlTemplateWatcher = htmlTemplateWatcher
+		if htmlTemplateWatcher, errorsChan := templates.NewWithWatcher(templatesDir); errorsChan != nil {
+			s.Go(func() {
+				for err2 := range errorsChan {
+					mlog.Error("Server templates error", mlog.Err(err2))
+				}
+			})
+		} else {
+			s.htmlTemplateWatcher = htmlTemplateWatcher
+		}
```

Comment:
I am not entirely sure if this is correct. In the normal case, `errorsChan` will not be nil. So it looks like we won't be assigning the `s.htmlTemplateWatcher` in that case.

I would say that there is no need to check for nil channel. The API _guarantees_ that the channel will always be non-nil. We can just range over it. The idea would be that whenever an error occurs, we close the watcher and set s.htmlTemplateWatcher to `nil` so that successive calls will fail.
```
var errChan chan error
s.htmlTemplateWatcher, errChan = templates.NewWithWatcher(dir)
s.Go(func() {
    err2 := <- range errorsChan 
    mlog.Error("Server templates error", mlog.Err(err2))
   s.htmlTemplateWatcher.Close()
  //some mutex
s.htmlTemplateWatcher = nil
 // end mutex
})
```

Reply:
:facepalm: